Tests/TestingApps/PaywallsTester/PaywallsTester/Configuration.swift

```diff
@@ -6,34 +6,69 @@
 //
 
 import Foundation
+import RevenueCat
 
-enum Configuration {
+final class Configuration: ObservableObject {
+    static let shared = Configuration()
 
-    #warning("Configure API key if you want to test paywalls from your dashboard")
-
-    // Note: you can leave this empty to use the production server, or point to your own instance.
-    static let proxyURL = ""
-    static let apiKey = ""
+    @Published private(set) var currentMode: Mode
 
     static let entitlement = "pro"
 
-}
-
-extension Configuration {
+    enum Mode {
+        case custom, testing, demos, listOnly
+    }
 
-    static var effectiveApiKey: String = {
-        return Self.apiKey.nonEmpty ?? Self.apiKeyFromCI
-    }()
+    #warning("Configure API key if you want to test paywalls from your dashboard")
+    // Note: you can leave this empty to use the production server, or point to your own instance.
+    private static let proxyURL = ""
+    private static let apiKey = ""
 
     // This is modified by CI:
-    private static let apiKeyFromCI = ""
+    private static let apiKeyFromCIForTesting = ""
+    private static let apiKeyFromCIForDemos = ""
 
-}
+    private init() {
+        self.currentMode = Self.apiKey.isEmpty ? .testing : .custom
```

Comment:
Should this be
```suggestion
        self.currentMode = Self.apiKey.isEmpty ? .listOnly : .custom
```

Reply:
good catch, this is wrong. It's slightly more complex in that there's 3 initial states, list only, testing and custom. It didn't jump out at me because in list only mode the configuration tab doesn't even show, but I'll fix it in any case